### PR TITLE
Integrational tests

### PR DESCRIPTION
**Pull request by PabloKarpacho**
Added 2 benchmarks and integrational tests with 4 scripts, covering all networks hyperparameters.

### CODE REVIEW

**Comment by jrzkaminski** [comment]
Since we have a `black` code style in the repo, please do 

```bash
pip install black
black tests/test_Integrational.py
```

File: tests/test_Integrational.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                if opt == 'HC':
+                    hybrid_network_params.append(
+                        (hybrid_data_path, mix, logit, opt, score, "Hybrid", 'target'))
+hybrid_network_params.append(
+    (hybrid_data_path, mix, logit, 'Evo', ("K2", K2Score), "Hybrid", 'target'))
+
+composite_network_params = [
+    (hybrid_data_path, False, False, 'Evo', ("K2", K2Score), "Composite", 'target')]
+
+params = disc_network_params + cont_network_params + hybrid_network_params
```
Comment (jrzkaminski) [inline tests/test_Integrational.py:52]: Looks better and shorter this way:

```suggestion
# Generate parameter lists using list comprehension and itertools.product
def generate_params(data_path, data_type, additional_params=None):
    params = []
    for opt, score in itertools.product(optimizer, scoring):
        if opt == 'HC':
            base_params = [data_path, False, False, opt, score, data_type]
            if additional_params:
                params.extend([base_params + list(param) for param in itertools.product(*additional_params)])
            else:
                params.append(base_params)
    # Add the 'Evo' optimizer case
    params.append([data_path] + [False, False, 'Evo', ("K2", K2Score), data_type] + (list(additional_params[0]) if additional_params else []))
    return params

disc_network_params = generate_params(discrete_data_path, "Discrete", [('Tectonic regime',)])
cont_network_params = generate_params(continuous_data_path, "Continuous", [(use_mixture, 'target')])
hybrid_network_params = generate_params(hybrid_data_path, "Hybrid", [(use_mixture, has_logit, 'target')])

composite_network_params = [
    (hybrid_data_path, False, False, 'Evo', ("K2", K2Score), "Composite", 'target')]

params = disc_network_params + cont_network_params + hybrid_network_params
```

**Review by jrzkaminski — CHANGES_REQUESTED** [review]
Please address issues mentioned above before merging

File: tests/test_Integrational.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
+import pytest
```
Comment (jrzkaminski) [inline tests/test_Integrational.py:1]: please also add this for the `generate_params` function to work

```python
import itertools 
```

**Review by jrzkaminski — CHANGES_REQUESTED** [review]

**Review by Roman223 — CHANGES_REQUESTED** [review]

**Review by Roman223 — APPROVED** [review]

**Review by Roman223 — APPROVED** [review]

**Review by jrzkaminski — APPROVED** [review]

**Pull request merged**